File: frontend_flutter_backup/lib/repositories/auth_repository.dart
import '../models/auth_session.dart';
import '../models/app_user.dart';
import '../services/api_service.dart';
import '../services/local_storage_service.dart';

class AuthRepository {
  AuthRepository(this.api, this.storage);
  final ApiService api;
  final LocalStorageService storage;
  Future<AuthSession> register({
    required String name,
    required String email,
    required String password,
    required String role,
  }) async {
    final r = await api.dio.post(
      '/auth/register',
      data: {
        'name': name,
        'email': email,
        'password': password,
        'role': role.toUpperCase(),
      },
    );
    final s = AuthSession.fromJson(r.data as Map<String, dynamic>);
    await storage.saveSession(s);
    return s;
  }

  Future<AuthSession> login({
    required String email,
    required String password,
  }) async {
    final r = await api.dio.post(
      '/auth/login',
      data: {'email': email, 'password': password},
    );
    final s = AuthSession.fromJson(r.data as Map<String, dynamic>);
    await storage.saveSession(s);
    return s;
  }

  Future<AuthSession> demoOauth(
    String provider, {
    String role = 'TRAVELER',
  }) async {
    final s = AuthSession(
      user: AppUser(
        id: 'demo-$provider',
        name: provider == 'google' ? 'Google User' : 'Apple User',
        email: 'you@${provider == 'google' ? 'gmail.com' : 'icloud.com'}',
        roles: [role == 'GUIDE' ? 'TRAVELER' : role],
      ),
      accessToken: 'demo-access',
      refreshToken: 'demo-refresh',
    );
    await storage.saveSession(s);
    return s;
  }

  Future<void> logout() async {
    final s = storage.readSession();
    if (s != null && !s.refreshToken.startsWith('demo')) {
      try {
        await api.dio.post(
          '/auth/logout',
          data: {'refreshToken': s.refreshToken},
        );
      } catch (_) {}
    }
    await storage.clearSession();
  }
}

File: frontend/src/repositories/authRepository.js
import { api } from '../services/api';
import { storage } from '../services/storage';
import { mapSession, mapUser } from '../models';

export const authRepository = {
  async register({ name, email, password, role }) {
    const { data } = await api.post('/auth/register', {
      name,
      email,
      password,
      role: String(role).toUpperCase(),
    });
    const session = mapSession(data);
    await storage.saveSession(session);
    return session;
  },

  async login({ email, password }) {
    const { data } = await api.post('/auth/login', { email, password });
    const session = mapSession(data);
    await storage.saveSession(session);
    return session;
  },

  async logout() {
    const session = storage.readSessionSync() || (await storage.readSession());
    if (session?.refreshToken) {
      try {
        await api.post('/auth/logout', {
          refreshToken: session.refreshToken,
        });
      } catch {
        // ignore
      }
    }
    await storage.clearSession();
  },

  async me() {
    const { data } = await api.get('/users/me');
    return mapUser(data);
  },

  async updateProfile(payload) {
    const { data } = await api.patch('/users/me', payload);
    return mapUser(data);
  },

  async verifyEmail(token) {
    const { data } = await api.post('/auth/verify-email', { token });
    return data;
  },

  async resendVerification(email) {
    const { data } = await api.post('/auth/resend-verification', { email });
    return data;
  },

  async forgotPassword(email) {
    const { data } = await api.post('/auth/forgot-password', { email });
    return data;
  },

  async resetPassword(token, newPassword) {
    const { data } = await api.post('/auth/reset-password', { token, newPassword });
    return data;
  },

  async changePassword(currentPassword, newPassword) {
    const { data } = await api.post('/auth/change-password', { currentPassword, newPassword });
    return data;
  },
};
